mlflow/xgboost/__init__.py

```diff
@@ -663,3 +483,232 @@ def infer_model_signature(input_example):
 
     safe_patch(FLAVOR_NAME, xgboost, "train", train, manage_run=True)
     safe_patch(FLAVOR_NAME, xgboost.DMatrix, "__init__", __init__)
+
+    # enable xgboost scikit-learn estimators autologging
+    import mlflow.sklearn
+
+    flavor_name_xgb_sklearn = "xgboost_sklearn"
+    xgb_sklearn_autolog = autologging_integration(flavor_name_xgb_sklearn)(mlflow.sklearn._autolog)
+    xgb_sklearn_autolog(
+        flavor_name=flavor_name_xgb_sklearn,
+        log_input_examples=log_input_examples,
+        log_model_signatures=log_model_signatures,
+        log_models=log_models,
+        disable=disable,
+        exclusive=exclusive,
+        disable_for_unsupported_versions=disable_for_unsupported_versions,
+        silent=silent,
+        max_tuning_runs=max_tuning_runs,
+        log_post_training_metrics=True,
+    )
```

Comment:
Instead of defining a new flavor name, can we pass in the `xgboost` flavor name to `mlflow.sklearn._autolog()`?

```suggestion
    # enable xgboost scikit-learn estimators autologging
    import mlflow.sklearn
    
    mlflow.sklearn._autolog(
        flavor_name=FLAVOR_NAME,
        log_input_examples=log_input_examples,
        log_model_signatures=log_model_signatures,
        log_models=log_models,
        disable=disable,
        exclusive=exclusive,
        disable_for_unsupported_versions=disable_for_unsupported_versions,
        silent=silent,
        max_tuning_runs=max_tuning_runs,
        log_post_training_metrics=True,
    )
```

If we use a new `xgboost_sklearn` flavor name, then disabling XGBoost autologging won't disable the XGBoost scikit-learn autologging patches, so we need to use the same flavor name.

Reply:
Thanks for the suggestion! I used `xgboost_sklearn` to avoid flavor conflict. Just tested using `xgboost`. Everything worked fine. Will remove `xgboost_sklearn` in future updates.